emily/handler/src/api/models/common/mod.rs

```diff
@@ -0,0 +1,79 @@
+//! Request structures for deposit api calls.
+
+use serde::{Deserialize, Serialize};
+use utoipa::{ToResponse, ToSchema};
+
+/// Common request structures.
+pub mod requests;
+
+// Primitives ------------------------------------------------------------------
+
+/// Type used to represent satoshis.
+pub type Satoshis = u128;
+/// Type used to represent a block hash.
+pub type StacksBlockHash = String;
+/// Type used to represent a block height.
+pub type BlockHeight = u64;
+/// Type used to represent a Bitcoin transaction Id.
+pub type BitcoinTransactionId = String;
+/// Type used to represent the output index on a Bitcoin transaction.
+pub type BitcoinTransactionOutputIndex = u32;
+/// Type used to represent a Stacks transaction Id.
+pub type StacksTransactionId = String;
+/// Type used to represent a bitcoin script.
+pub type BitcoinScript = String;
+/// Type used to represent a Stacks address principle.
+pub type StacksPrinciple = String;
+/// Type used to represent a Bitcoin address.
+pub type BitcoinAddress = String;
```

Comment:
I like that the aliases encode the meaning of the fields and make it easier to update. For example, The `BitcoinAddress` and the `BlockHash` may change to being a `Vec<u8>` depending on whether I get to changing it, and the alternative is that I make that edit a lot of places and hope I hit all of them.

I see your point, I think I might need to do that anyway because of the schema issue, but for now would you mind if I added a todo and kept this? I think there's some chance these types change their under-the-hood representation when I'm managing its interactions with `dynamodb` to be as seemless as possible.

Reply:
I also like the type aliases. They convey more relevant information than just the raw types. It also paves the way for easier refactors if we want to change the types to more complex ones.

Reply:
I'm more than okay with type aliases on as we're building things out, they make sense there too (something I neglected to mention above), but they don't make sense once things become a little more mature.

---

If we want to convey more information, I'd prefer using the [newtype](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) pattern instead. I'm much more in favor of that pattern. Type aliases don't enforce any of the guarantees that we'd like, as you're free to mix and match however you like. For example, the following is perfectly fine:
```rust
type BitcoinTx = Vec<u8>;
type BitcoinTxId = Vec<u8>;
type StacksTxId = Vec<u8>;

trait DbRead {
    async fn get_bitcoin_tx(&self, block_hash: &BitcoinTxId) -> Result<BitcoinTx, Error>;
}

impl DbRead for () {
    /// This compiles perfectly fine
    async fn get_bitcoin_tx(&self, block_hash: &StacksTxId) -> Result<BitcoinTxId, Error> {
        Ok(Vec::new())
    }
}
```

Reply:
It might not be a choice  because the way `utoipa` makes schemas doesn't support these aliases. These API tools really force our hand on some structures.

That said, on the attitute towards type aliases, the above code doesn't seem like a problem to me.

Reply:
> It might not be a choice because the way `utoipa` makes schemas doesn't support these aliases. These API tools really force our hand on some structures.

Oh what do you mean? As in `utoipa` doesn't pick up on the type alias at all?

Reply:
> If we want to convey more information, I'd prefer using the [newtype](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) pattern instead.

100% agree that newtypes are better than type aliases, though type aliases is a strict improvement over just using the raw types directly in my view. The way I see it is that most type aliases should be migrated to newtypes as we evolve the code. That seems to be what you're getting at too @djordon.

Reply:
> The way I see it is that most type aliases should be migrated to newtypes as we evolve the code. That seems to be what you're getting at too @djordon.

Yes, 100%